airflow/jobs/scheduler_job.py

```diff
@@ -1292,13 +1290,12 @@ def _execute(self) -> None:
                 )
                 models.DAG.deactivate_stale_dags(execute_start_time)
 
-            self.executor.end()
-
             settings.Session.remove()  # type: ignore
         except Exception:  # pylint: disable=broad-except
             self.log.exception("Exception when executing SchedulerJob._run_scheduler_loop")
         finally:
             self.processor_agent.end()
```

Comment:
what if `self.processor_agent.end()` fails? will `self.executor.end()` be reached?  maybe you need something like an exit stack

Reply:
Good point. I tried exit stack but wasn't able to get it to log any exceptions, if any happened, in the end methods. I fell back to try/catch, but open to ideas.